adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -0,0 +1,477 @@
+        validator=instance_of(OntologyLexicon), kw_only=True
+    )
+
+    # how to assign word order to the dependency trees
+    _dependency_tree_linearizer: DependencyTreeLinearizer = attrib(
+        init=False, default=SIMPLE_CHINESE_DEPENDENCY_TREE_LINEARIZER, kw_only=True
+    )
+
+    # the function that actually generates the language
+    def generate_language(
+        self, situation: HighLevelSemanticsSituation, chooser: SequenceChooser
+    ) -> ImmutableSet[LinearizedDependencyTree]:
+        # remove once done
+        return SimpleRuleBasedChineseLanguageGenerator._Generation(
+            self, situation
+        ).generate()
+
+    """This class encapsulates all the mutable state for an execution of the 
+    SimpleRuleBasedChineseLanguageGenerator on a single input"""
+
+    @attrs(frozen=True, slots=True)
+    class _Generation:
+        # keep the reference to the parent because python doesn't have real inner classes
+        generator: "SimpleRuleBasedChineseLanguageGenerator" = attrib()
+        # the situation being translated into language
+        situation: HighLevelSemanticsSituation = attrib()
+        # the graph we are building
+        dependency_graph: DiGraph = attrib(init=False, default=Factory(DiGraph))
+        # stores a mapping of nouns for the objects in the situation
+        objects_to_dependency_nodes: MutableMapping[
+            SituationObject, DependencyTreeToken
+        ] = attrib(init=False, factory=dict)
+        # keep a mapping of object counts so we know what quantifiers to use when there are multiple objects
+        object_counts: Mapping[OntologyNode, int] = attrib(init=False)
+
+        """The function that tries to generate language for a given representation"""
+
+        def generate(self) -> ImmutableSet[LinearizedDependencyTree]:
+            try:
+                return self._real_generate()
+            except Exception as e:
+                raise RuntimeError(
+                    "Error while generating Chinese for the situation {}".format(
+                        self.situation
+                    )
+                ) from e
+
+        """The function that actually generates the language"""
+
+        def _real_generate(self) -> ImmutableSet[LinearizedDependencyTree]:
+            # we currently can't deal with situations with more than one action
+            if len(self.situation.actions) > 1:
+                raise RuntimeError(
+                    "Currently only situations with 0 or 1 actions are supported"
+                )
+
+            # handle the special case of a static situation with only multiple objects of the same type
+            object_types_in_situation = set(
+                object_.ontology_node for object_ in self.situation.salient_objects
+            )
+
+            action: Optional[Action[OntologyNode, SituationObject]]
+            # handle dynamic situations
+            if self.situation.is_dynamic:
+                raise NotImplementedError
```

Comment:
```suggestion
                raise NotImplementedError()
```

Reply:
Otherwise what you raise is not an instance of `NotImplementedError`, but the class itself.